anonlink/concurrency.py

```diff
@@ -86,3 +91,79 @@ def split_to_chunks(
                 _chunks_1d(size0, chunks0), _chunks_1d(size1, chunks1)):
             yield [{'datasetIndex': i0, 'range': c0},
                    {'datasetIndex': i1, 'range': c1}]
+
+
+def _get_dataset_indices(dataset_chunk, size):
+    index = dataset_chunk['datasetIndex']
+    return _array.array('I', (index,)) * size
+
+
+def _offset_record_indices(dataset_chunk, rec_is):
```

Comment:
Minor thing: `_offset_record_indices` modifies `rec_is` _in-place_, contrary to most other functions here and elsewhere in the project (which return new values, making them more obviously referentially transparent). Given that the only uses of this function and the one above are together (line 163ff), and given that they are so short, you could easily combine them to get a referentially transparent function.

Reply:
Ignore the comment above; I mistakenly thought the output of `_get_dataset_indices` was being passed directly to `_offset_record_indices`, but that's not the case.

Perhaps still a good idea to make it clear that `_offset_record_indices` modifies its parameter in place.